### ISSUE TITLE
asdf install kubectl latest:<major.minor> broken

### ISSUE THREAD

**Issue by allen-servedio** [Banno/asdf-kubectl#28]
It appears that #26 broke installing the latest version that begins with a string (so calling something like: `asdf install kubectl latest:<major.minor>`:

Previous version of the plugin behavior (I had run `asdf install kubectl latest` previous to the command below to install 1.23.4):
```
> asdf plugin update kubectl 711ad7d
Updating kubectl to 711ad7d
fatal: couldn't find remote ref 711ad7d
HEAD is now at 711ad7d Merge pull request #23 from Savemech/patch-1

> asdf install kubectl latest:1.22
Downloading kubectl from https://dl.k8s.io/release/v1.22.7/bin/darwin/amd64/kubectl

> asdf list kubectl
  1.22.7
  1.23.4
```

Current version of the plugin:
```
> asdf plugin update kubectl
Updating kubectl to master
Previous HEAD position was 711ad7d Merge pull request #23 from Savemech/patch-1
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

> asdf install kubectl latest:1.22
kubectl 1.23.4 is already installed
```

And just in case that was misleading:
```
> asdf list kubectl
  1.22.7
  1.23.4

> asdf uninstall kubectl 1.22.7

> asdf list kubectl
  1.23.4

> asdf install kubectl latest:1.22
kubectl 1.23.4 is already installed

> asdf list kubectl
  1.23.4
```

As you can see, it no longer installs the latest version matching the string passed in.

**Comment by allen-servedio** [comment]
FYI @log2 - maybe an unintended side effect?

**Comment by allen-servedio** [comment]
Okay @log2 and @DustinChaloupka - I think I see what was missed from #26 (from http://asdf-vm.com/plugins/create.html#optional-scripts):

> This callback is invoked with a single "filter" string as it's only argument. This should be used for filter all latest stable versions. For example with Ruby, the user may choose to pass in jruby to select the latest stable version of jruby.

So in our case, when we call `asdf install kubectl latest:1.22` it should filter the versions to return the latest 1.22.x version. With #26 it ignores the filter and just returns the latest version of `kubectl`.

**Comment by log2** [comment]
Thank you for spotting this @allen-servedio , totally missed the optional argument. I will try to fix this with a more comprehensive patch. However, in order to preserve plugin's behaviour for all users for the time being, could you please fix this, at least temporarily, by reverting my previous patch? Thanks

**Comment by log2** [comment]
Filtering of latest versions is provided by [this line](https://github.com/asdf-vm/asdf/blob/master/lib/commands/command-latest.bash#L31), implementing the same filtering on latest-stable script, when parameter is provided, should restore previous behaviour. Working on it.

**Comment by allen-servedio** [comment]
@log2 - I am only a user of this plugin so I can't revert anything in this repo.... However @DustinChaloupka should be able to revert the last commit. 

Thanks for jumping on this so quickly @log2 ! Once we realized what had happened, we pinned the plugin in our automation to an older version so that our users were not impacted until this is fixed or rolled back. Thanks again - I am sure your change will ultimately make this work well for the future (since you are quite correct that there will be many versions of that cli).
